### ISSUE TITLE
FABButton depth preset stop working

### ISSUE THREAD

**Issue by StepanetsDmtry** [CosmicMind/Material#1110]
After last update 2.16.0 a `depthPreset` property stop working for circle `shapePreset`. This is because after installing a `depthPreset ` `CALayer` calls `layoutShape()` method. In this method, `maskToBounds` property always is true for circle `shapePreset`. If I change `maskToBounds = false`, all working fine.

```
open func layoutShape() {
    guard .none != shapePreset else {
      return
    }
    
    if 0 == bounds.width {
      bounds.size.width = bounds.height
    }
    
    if 0 == bounds.height {
      bounds.size.height = bounds.width
    }
    
    guard .circle == shapePreset else {
      masksToBounds = false
      cornerRadius = 0
      return
    }
    
    masksToBounds = true
    cornerRadius = bounds.width / 2
  }
```

**Comment by OrkhanAlikhanov** [comment]
@danieldahan I confirm that `FABButton` has no depth after https://github.com/CosmicMind/Material/commit/b90ee8ac753685ba5027e157449dde7c5ebf85bb as explained in the issue.

Basically we have no control over `layer.masksToBounds` property as it's overridden on layout cycle. Only workaround is to subclass `FABButton` (or any affected component) and change the property there.
```swift
class FABButton: Material.FABButton {
  override func layoutSubviews() {
    super.layoutSubviews()
    layer.masksToBounds = false
   }
}
```

**Comment by danieldahan** [comment]
`masksToBounds = true` should actually be removed. It is incorrectly being placed on the wrong layer. It should be added to the `visualLayer`. I will fix this as I know the complexity of its behaviour well. I will work on this tonight. Thank you!

**Comment by danieldahan** [comment]
Fixed the issue f99b6dfdc61e99d8bf30168a4262383b8c184576, and will be released in a few hours. Thank you!
